### PR TITLE
Model

### PR DESCRIPTION
**Pull request by JaneIllario**
preliminary model implementation - CNN for non-sequential data

### CODE REVIEW

File: data_processing.ipynb [hunk 1]
```diff
@@ -30,7 +30,66 @@
...
+      "Requirement already satisfied: numpy>=1.15 in /usr/local/lib/python3.7/site-packages (from matplotlib) (1.19.1)\n",
+      "Requirement already satisfied: python-dateutil>=2.1 in /usr/local/lib/python3.7/site-packages (from matplotlib) (2.8.1)\n",
+      "Requirement already satisfied: kiwisolver>=1.0.1 in /usr/local/lib/python3.7/site-packages (from matplotlib) (1.3.1)\n",
+      "Requirement already satisfied: cycler>=0.10 in /usr/local/lib/python3.7/site-packages (from matplotlib) (0.10.0)\n",
+      "Requirement already satisfied: six>=1.5 in /usr/local/lib/python3.7/site-packages (from python-dateutil>=2.1->matplotlib) (1.14.0)\n"
+     ]
+    }
+   ],
+   "source": [
+    "!pip3 install matplotlib"
```
Comment (eanile) [inline data_processing.ipynb]: Is there a reason why we're installing packages in the notebook? I included a requirements.txt for a reason, so they can all be installed from there

File: data_processing.ipynb [hunk 2]
```diff
@@ -113,11 +172,24 @@
...
-   "execution_count": 38,
+   "execution_count": 12,
    "metadata": {
     "tags": []
    },
-   "outputs": [],
+   "outputs": [
+    {
+     "ename": "FileNotFoundError",
+     "evalue": "[Errno 2] No such file or directory: 'data/TimeSeries'",
```
Comment (eanile) [inline data_processing.ipynb]: Why is this error occuring? Do you not have the data downloaded?

File: data_processing.ipynb [hunk 3]
```diff
@@ -748,6 +820,726 @@
...
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 24,
+   "metadata": {},
+   "outputs": [
+    {
+     "ename": "KeyboardInterrupt",
+     "evalue": "",
+     "output_type": "error",
```
Comment (eanile) [inline data_processing.ipynb]: Keyboard interrupt? All this error code should not be in a PR

File: data_processing.ipynb [hunk 4]
```diff
@@ -748,6 +820,726 @@
...
+    "\n",
+    "directory = 'data/images/Baseline'\n",
+    "for filename in os.listdir(directory):\n",
+    "    if filename.endswith(\".png\"):\n",
+    "        img = img_to_array(load_img(os.path.join(directory, filename)))\n",
+    "        data.append(img)\n",
+    "        y=0\n",
+    "        labels.append(y)\n",
+    "    else:\n",
+    "        continue\n",
```
Comment (eanile) [inline data_processing.ipynb]: This code is essentially repeated 4 times, for each different directory/label. Please either make a function for it or write the loop once instead of repeating it.

File: data_processing.ipynb [hunk 5]
```diff
@@ -748,6 +820,726 @@
...
+    "print(\"Shape before one-hot encoding: \", y_train.shape)\n",
+    "Y_train = np_utils.to_categorical(y_train, n_classes)\n",
+    "Y_test = np_utils.to_categorical(y_test, n_classes)\n",
+    "print(\"Shape after one-hot encoding: \", Y_train.shape)\n",
+    "\n",
+    "# building a linear stack of layers with the sequential model\n",
+    "model = Sequential()\n",
+    "# convolutional layer\n",
+    "model.add(Conv2D(25, kernel_size=(3,3), strides=(1,1), padding='valid', activation='relu', input_shape=(313,217,3)))\n",
+    "model.add(MaxPool2D(pool_size=(1,1)))\n",
```
Comment (eanile) [inline data_processing.ipynb]: Only a 1x1 pooling?

File: data_processing.ipynb [hunk 6]
```diff
@@ -748,6 +820,726 @@
...
+    "# hidden layer\n",
+    "model.add(Dense(100, activation='relu'))\n",
+    "# output layer\n",
+    "model.add(Dense(4, activation='softmax'))\n",
+    "\n",
+    "# compiling the sequential model\n",
+    "model.compile(loss='categorical_crossentropy', metrics=['accuracy'], optimizer='adam')\n",
+    "\n",
+    "# training the model for 10 epochs\n",
+    "history = model.fit(X_train, Y_train, batch_size=128, epochs=20, validation_data=(X_test, Y_test))\n"
```
Comment (eanile) [inline data_processing.ipynb]: Seems that here we only have train and test data. The test data is really being used as validation data. But really, we should have train, test, and validation data.

**Review by eanile — CHANGES_REQUESTED** [review]
General comment about this PR. This entire .ipynb file is supposed to be only for data processing. I think we should remove the model code and separate that into another notebook. We should be able to import the data processing functions in this notebook into that and other notebooks. If this is not possible, perhaps we should also think about converting this notebook to a python script that defines all the functions. We may want to keep both the notebook and the script, as the notebook gives us a visual representation of outputs of these functions, but we also don't want to repeat ourselves.

**Comment by JaneIllario** [comment]
Updated as per the main comment -- notebook is no longer for data processing but the whole project as is reflected by capstone title. Sections are demarcated with new markup headers to separate data processing from model, and model data aggregation. All other comments were relevant to previous edits on the notebook (not specifically from this PR) and were handled by a merge with the updated master branch.

File: data_processing.ipynb (hunk 6 above)
Comment (JaneIllario) [reply data_processing.ipynb]: Validation testing is not a necessary indicator; sufficiently allocated test data will suffice for the baseline. The split written by Rea was not used and can be deleted; it would make more sense to alter the model here if the need arises.
